packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -657,6 +662,15 @@ class _DatePickerDialogState extends State<_DatePickerDialog> {
 
   @override
   Widget build(BuildContext context) {
+    switch (Theme.of(context).platform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+        _vibrate = true;
+        break;
+      case TargetPlatform.iOS:
+        _vibrate = false;
+        break;
+    }
```

Comment:
> For my own education, what's the difference?

There's very little operational difference.  The general pattern is that build() functions should be free of side effects, and writing the _vibrate field is a side effect, so it's better code hygiene to do it the other way.

> This only changes when set state is called I assume?

If you're called Theme.of(context) since the last build, then build() will be called if the theme data changes.  (Technically it uses a lower-level mechanism than setState, but the effect is the same.)

> If it were possible to keep the modal dialog on while changing the platform in the gallery, would it indirectly trigger a set state or no?

That is possible.  There's a command in the hot reload command line tool to flip the target platform, which you can do while the dialog is on screen.  (Or the all can just do it on a timer whenever it likes.)

When that happens, that will trigger build() to be called again if that widget has called Theme.of(context) since the last time the framework called build().

Basically, we keep track of all the inherited widgets that you've read since build() and make sure to call build() again if any of them change.  If you don't read from an inherited widget the second time around, we won't call build() again when it changes.

In summary, the code in this patch is correct.  It just violates a code hygiene/style guideline.  We try to make the framework code as clean as possible under the assumption that many people will use the framework code as a template for their own code, and it would be better if they copy/pasted the other pattern.